pytorch_lightning/utilities/apply_func.py

```diff
@@ -257,8 +258,11 @@ def move_data_to_device(batch: Any, device: Union[str, torch.device]) -> Any:
 
     def batch_to(data: Any) -> Any:
         # try to move torchtext data first
-        if _TORCHTEXT_AVAILABLE and isinstance(data, Batch):
-
+        if _TORCHTEXT_LEGACY and isinstance(data, Batch):
+            rank_zero_warn(
+                "The torchtext.legacy.Batch object is deprecated and Lightning will remove support for it in v1.8."
+                "We recommend you to migrate away from Batch by following the TorchText README: https://github.com/pytorch/text#bc-breaking-legacy"
+            )
```

Comment:
Can you add a test for this message in https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-8.py?